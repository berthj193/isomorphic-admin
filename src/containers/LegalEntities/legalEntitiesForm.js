import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import Form, { FormSubmit } from '../../components/uielements/form';
import Input from '../../components/uielements/input';
import { required } from '../../helpers/formValidation';
import withReduxForm from '../../components/formWrapper/withReduxForm';

export class LegalEntityForm extends React.Component {
  static propTypes = {
    items: PropTypes.string,
    handleSubmit: PropTypes.func,
  };

  render() {
    return (
      <Form onSubmit={this.props.handleSubmit}>
        <Field label="Name" name="name" component={Input} validate={[required]} required />
        <Field label="VAT Number" name="vat_number" component={Input} />

        <FormSubmit />
      </Form>
    );
  }
}

export default withReduxForm({
  form: 'legalEntitiesForm',
  enableReinitialize: true,
})(LegalEntityForm);
