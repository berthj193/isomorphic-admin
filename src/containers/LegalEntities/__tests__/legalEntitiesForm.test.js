import React from 'react';
import { LegalEntityForm } from '../legalEntitiesForm';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<LegalEntityForm />', () => {
  const getComponent = (props = {}) => shallow(<LegalEntityForm {...props} />);

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });
});
