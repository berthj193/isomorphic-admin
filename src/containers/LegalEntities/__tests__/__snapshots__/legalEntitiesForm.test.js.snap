// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegalEntityForm /> should render properly 1`] = `
<Form
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
  prefixCls="ant-form"
>
  <Field
    component={[Function]}
    label="Name"
    name="name"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    label="VAT Number"
    name="vat_number"
  />
  <FormSubmit />
</Form>
`;
